R/find.tol.time.R

```diff
@@ -1,86 +1,102 @@
-find.tol.time <-
-function(mz, chr, lab, num.exp, mz.tol=2e-5, chr.tol=NA, aver.bin.size=200, min.bins=50, max.bins=100, max.mz.diff=0.01, max.num.segments=10000, do.plot=TRUE)
-{
-    o<-order(mz)
-    mz<-mz[o]
-    chr<-chr[o]
-    lab<-lab[o]
+find.tol.time <- function(mz,
+                          chr,
+                          lab,
+                          number_of_experiments,
+                          mz_tol_relative = 2e-5,
+                          rt_tol_relative = NA,
+                          aver.bin.size = 200,
+                          min.bins = 50,
+                          max.bins = 100,
+                          mz_tol_absolute = 0.01,
+                          max.num.segments = 10000,
+                          do.plot = TRUE) {
+    o <- order(mz)
+    mz <- mz[o]
+    chr <- chr[o]
+    lab <- lab[o]
     rm(o)
     
-    l<-length(mz)
+    l <- length(mz)
     
-    breaks<-c(0, which((mz[2:l]-mz[1:(l-1)]) > min(max.mz.diff, mz.tol*((mz[2:l]+mz[1:(l-1)])/2))), l)
+    breaks <-
+        c(0, which((mz[2:l] - mz[1:(l - 1)]) > min(mz_tol_absolute, mz_tol_relative * ((
+            mz[2:l] + mz[1:(l - 1)]
+        ) / 2))), l)
     
-    for(i in 2:length(breaks))
-    {
-        this.o<-order(chr[(breaks[i-1]+1):breaks[i]])
-        this.o<-this.o+breaks[i-1]
-        mz[(breaks[i-1]+1):breaks[i]]<-mz[this.o]
-        chr[(breaks[i-1]+1):breaks[i]]<-chr[this.o]
-        lab[(breaks[i-1]+1):breaks[i]]<-lab[this.o]
+    for (i in 2:length(breaks)) {
+        this.o <- order(chr[(breaks[i - 1] + 1):breaks[i]])
+        this.o <- this.o + breaks[i - 1]
+        mz[(breaks[i - 1] + 1):breaks[i]] <- mz[this.o]
+        chr[(breaks[i - 1] + 1):breaks[i]] <- chr[this.o]
+        lab[(breaks[i - 1] + 1):breaks[i]] <- lab[this.o]
```

Comment:
this also seems familiar - check out the feature extraction functions, there is likely common functionality